bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableScanConfiguration.java

```diff
@@ -188,15 +252,12 @@ protected CloudBigtableScanConfiguration(
       ValueProvider<String> projectId,
       ValueProvider<String> instanceId,
       ValueProvider<String> tableId,
-      ReadRowsRequest request,
+      ValueProvider<ReadRowsRequest> request,
       Map<String, ValueProvider<String>> additionalConfiguration) {
     super(projectId, instanceId, tableId, additionalConfiguration);
     this.request =
         NestedValueProvider.of(
-            // Eventually the input request will be ValueProvider<ReadRowsRequest>.
-            // TODO(kevinsi): Make sure that the resulting request object is accessible only when
-            // all dependent runtime parameters are accessible.
-            StaticValueProvider.of(request),
+            request,
```

Comment:
I think Igor has concern here on whether we should check to make sure all request, project ID and instance ID must be accessible. I wonder if one of them is not accessible, how do we handle it? Can we just let it fail when one of the get() is called?

Reply:
Added RequestWithTableNameValueProvider.